Python/pywarpx/fields.py

```diff
@@ -544,206 +536,246 @@ def max_index(self, *args):
     def norm0(self, *args):
         return self.mf.norm0(*args)
 
+
+class _WarpXMultiFABWrapper(_MultiFABWrapper):
+    """Wrapper around WarpX MultiFabs
+    This provides a convenient way to query and set data in the MultiFabs.
+    The indexing is based on global indices.
+
+    Parameters
+    ----------
+     mf_name: string
+         The name of the MultiFab to be accessed, the tag specified when the
+         MultiFab is allocated
+
+     level: int
+         The refinement level
+
+     include_ghosts: bool, default=False
+         Whether to include the ghost cells.
+         Note that when True, the first n-ghost negative indices will refer to the lower
+         ghost cells.
+    """
+    def __init__(self, mf_name, level, include_ghosts=False):
+        self.mf_name = mf_name
+        self.level = level
+        self.include_ghosts = include_ghosts
+
+        self.dim = libwarpx.dim
+
+        # The overlaps list is one along the axes where the grid boundaries overlap the neighboring grid,
+        # which is the case with node centering.
+        ix_type = self.mf.box_array().ix_type()
+        self.overlaps = self._get_indices([int(ix_type.node_centered(i)) for i in range(self.dim)], 0)
+
+    @property
+    def mf(self):
+        # Always fetch this anew in case the C++ MultiFab is recreated
+        warpx = libwarpx.libwarpx_so.get_instance()
+        # All MultiFab names have the level suffix
+        return warpx.multifab(f'{self.mf_name}[level={self.level}]')
+
+
 def ExWrapper(level=0, include_ghosts=False):
-    return _MultiFABWrapper(mf_name=f'Efield_aux[x]', level=level, include_ghosts=include_ghosts)
+    return _WarpXMultiFABWrapper(mf_name=f'Efield_aux[x]', level=level, include_ghosts=include_ghosts)
```

Comment:
While you are making changes here, I notice that the `f` before the string is not necessary. It is leftover from a previous version that did need it and can now be removed.